### PR TITLE
AzureCommunicationChat GA Feedback

### PR DESCRIPTION
**Pull request by knvsl**
Addressing some of the comments here: https://github.com/Azure/azure-sdk-for-ios/pull/851
New swiftinterface based off these changes here: https://github.com/Azure/azure-sdk-for-ios/pull/925

- [X] Update trouter classes to structs
- [X] Add missing docs for public methods
- [X] Update Trouter package to beta5 (tested in signaling app)
- [X] Review things that are public if they should be internal (IdentifierSerializer is internal)
- [X] Review naming consistency - request, options, result
- [X] Document M1 support (Added note in README that references #787 )

### CODE REVIEW

**Comment by knvsl** [comment]
@tjprescott looks like Smoketest is failing because it can't find latest trouter package - is there anything I need to change in the smoketest example?

File: sdk/communication/AzureCommunicationChat/AzureCommunicationChat.podspec.json [hunk 1]
```diff
@@ -35,8 +35,8 @@
     "AzureCommunicationCommon": [
       "~> 1.0"
     ],
-    "TrouterClientIos": [
-      "0.0.1-beta.4"
+    "TrouterClient": [
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationChat/AzureCommunicationChat.podspec.json:38]: This should be `Trouter` not `TrouterClient`

File: sdk/communication/AzureCommunicationChat/AzureCommunicationChat.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1309,7 +1311,7 @@
 			repositoryURL = "https://github.com/microsoft/trouter-client-ios/";
 			requirement = {
 				kind = upToNextMajorVersion;
-				minimumVersion = "0.0.1-beta.4";
+				minimumVersion = "0.0.1-beta.5";
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationChat/AzureCommunicationChat.xcodeproj/project.pbxproj:1314]: I think this is a branch, not a tag, according to Jiri.

**Review by tjprescott — COMMENTED** [review]
Overall LGTM

File: sdk/communication/AzureCommunicationChat/AzureCommunicationChat.podspec.json (hunk 1 above)
Comment (knvsl) [reply sdk/communication/AzureCommunicationChat/AzureCommunicationChat.podspec.json]: Fixed, thanks

File: sdk/communication/AzureCommunicationChat/AzureCommunicationChat.xcodeproj/project.pbxproj (hunk 2 above)
Comment (knvsl) [reply sdk/communication/AzureCommunicationChat/AzureCommunicationChat.xcodeproj/project.pbxproj:1314]: its tagged!

**Comment by JoshuaLai** [comment]
Changes look good to me!

**Review by juancamilor — APPROVED** [review]

**Pull request merged**